src/nvim/edit.c

```diff
@@ -7013,21 +7013,23 @@ static void ins_reg(void)
   }
 
 
-  /*
-   * Don't map the register name. This also prevents the mode message to be
-   * deleted when ESC is hit.
-   */
-  ++no_mapping;
-  regname = plain_vgetc();
-  LANGMAP_ADJUST(regname, TRUE);
-  if (regname == Ctrl_R || regname == Ctrl_O || regname == Ctrl_P) {
-    /* Get a third key for literal register insertion */
-    literally = regname;
-    add_to_showcmd_c(literally);
+  // Don't map the register name.
+  // This also prevents the mode message to be deleted when ESC is hit.
+  // Also, pretend it's normal mode so that LANGMAP is applied.
```

Comment:
I think the first two lines belong above `no_mapping++` (as they were), and the last line should be next to `State = NORMAL;`.